[docs] Update README.md (#97)
MystenLabs/seal: README.md

We strive to maintain backward compatibility for Seal, and any breaking changes to the Key Server API and the TypeScript SDK will be communicated in advance both on GitHub and in the Seal Discord channel.

> We will also provide a migration path for existing users to upgrade to the new version.

Reviewer: ```suggestion
We will provide a clear migration path for existing users to upgrade to new versions.
```

> New versions of the Key Server will be published on GitHub under [Releases](https://github.com/MystenLabs/seal/releases) and will be tagged with the version number.

Reviewer: ```suggestion
New versions of the Key Server will be published under [GitHub Releases](https://github.com/MystenLabs/seal/releases) and tagged with the appropriate version number.
```

> Note that both the [SDK](https://github.com/MystenLabs/ts-sdks/blob/main/packages/seal/src/key-server.ts#L31) and the [Key Server](https://github.com/MystenLabs/seal/blob/main/crates/key-server/src/server.rs#L85) are versioned, and may reject messages from outdated or deprecated counterparties.

Reviewer: ```suggestion
Both the [SDK](https://github.com/MystenLabs/ts-sdks/blob/main/packages/seal/src/key-server.ts#L31) and the [Key Server](https://github.com/MystenLabs/seal/blob/main/crates/key-server/src/server.rs#L85) are configured to require specific versions of each other. Either component may reject messages from outdated or deprecated counterparties.
```

> Note that both the SDK and the Key Server are versioned, and may reject messages from outdated or deprecated counterparties (see minimal versions in [SDK](https://github.com/MystenLabs/ts-sdks/blob/main/packages/seal/src/key-server.ts#L31) and [Key Server](https://github.com/MystenLabs/seal/blob/main/crates/key-server/src/server.rs#L85)).

Reviewer: ```suggestion
Both the [SDK](https://github.com/MystenLabs/ts-sdks/blob/main/packages/seal/src/key-server.ts#L31) and the [Key Server](https://github.com/MystenLabs/seal/blob/main/crates/key-server/src/server.rs#L85) are configured to expect specific versions from each other. Either component may reject messages from outdated or deprecated counterparties.
```
